build-support/bump_version.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import print_function
+
+import argparse
+import logging
+import os
+import sys
+
+from semver_utils import SemanticVersionComponent, bump_version_component
+from version_file import read_version
+from version_writer import VersionWriter
+
+
+class BumpVersionCLI:
+
+    LOG_FORMAT = "%(asctime)s:%(levelname)s:%(name)s:%(message)s"
+
+    class ArgumentError(RuntimeError):
+        pass
+
+    def __init__(self):
+        self._parser = None
+        self._command_name = "version_bumper"
+
+        self.log_level = None
+        self.new_sdk_version = None
+        self.root_dir = None
+
+    def run(self, argv):
+        self._command_name = os.path.basename(argv[0])
+        args = self.parse_args(argv[1:])
+        self.process_args(args)
+
+        logging.debug(f"new_sdk_version: {self.new_sdk_version}, root_dir: {self.root_dir}")
+
+        self.write_new_version()
+
+    def setup_parser(self):
+        self._parser = argparse.ArgumentParser(
+            description="Bump version of the SDK",
+            usage=f"{self._command_name} <args>",
+        )
+
+        # NOTE: The "choices" allowed by this argument must exactly match the values of
+        # the SemanticVersionComponent enum
+        self._parser.add_argument(
+            "-c",
+            "--component",
+            help="specify the version component (major, minor, patch) to bump",
+            choices=["MAJOR", "MINOR", "PATCH"],
```

Comment:
question: is it possible to use a list comprehension to generate these values? i.e. `[component.value for component in SemanticVersionComponent]`

Reply:
Good callout! Will do that in a future PR.